### PR TITLE
AP_GPS: ublox raise baud rate and allow user controlled baud rates

### PR DESCRIPTION
**Pull request by WickedShell**
Raise the default baudrate to 115200 for all u-blox devices. This was tested primarily 6-8 series devices on the ground. I don't have a 4-5 series device to test with.

Also addresses seeing a 5Hz warning on initial gain of GPS fix if it took more then 20 seconds after detecting the GPS unit to have a fix.

### CODE REVIEW

**Comment by WickedShell** [comment]
Tagging as need rework at the moment as this might as well not go in until all the other update rate changes are put together, as that is going to squash the 5hz stuff from this anyways

**Comment by WickedShell** [comment]
Actually inclined to delete the 5Hz warning from the u-blox driver entirely. It's entire purpose was to try and detect a brownout of the GPS, but we cover that scenario much better now that the driver is auditing the entire part of the configuration we care about. We are more likely to detect a problem this way, and we have a very reasonable chance of detecting it earlier then waiting 20 seconds for the 5Hz check to come around.

The 5Hz warning currently generates a fairly large level of noise as loosing fix for >20 seconds can also trigger it, but since this is mostly on ground testing users are trained to ignore the warning. I'm inclined to ensure that the current state of the ublox arming checks is logged on one of the UBX messages, which will allow us to detect the brownout/configuration problem when reviewing logs. (Alternatively if we have fully configured and confirmed the configuration and anything has become unset I could then broadcast that to the user instead rather then the 5Hz warning) but this is far better coverage then just the update rate when diagnosing GPS problems.

Any objections to deleting the current form of the warning? @tridge @rmackay9 @lvale @peterbarker

**Comment by magicrub** [comment]
>this might as well not go in until all the other update rate changes are put together
https://github.com/ArduPilot/ardupilot/pull/4884 has been applied, is that what you are referring to?

also, this has merge conflicts now

**Comment by WickedShell** [comment]
@magicrub no was referring to #3717 Yeah, the merge conflicts came from my other PR as well, so that's not a surprise :)

**Comment by magicrub** [comment]
@WickedShell What's the status of this, still want it in? Can you fix the merge conflict?

**Comment by magicrub** [comment]
This is waiting for other changes, should have it in a few days (per @WickedShell 's comment on the recent dev call)

**Comment by WickedShell** [comment]
_DO NOT MERGE YET_ (Although feel free to test, it should be stable with sane parameters)

I've removed the 5Hz warning entirely in favor of the other checks which are already auditing the same settings in a more proactive manner.

This also allows controlling the update rate in milliseconds. I've set a minimum update rate of 5Hz, however I haven't limited the maximum update rate. I've only tested on u-blox 8th gen devices at the moment, will test earlier hardware generations tomorrow.

The larger concern is that I need to validate what happens when a user requests a update rate that is to fast/out of range. At first glance so far it appears to handle it in a sane and safe manner, but I need to ensure that is the case before we can think about merging this.

File: libraries/AP_GPS/AP_GPS_UBLOX.cpp [hunk 1]
```diff
@@ -1144,7 +1144,8 @@ void
 AP_GPS_UBLOX::_configure_rate(void)
 {
     struct ubx_cfg_nav_rate msg;
-    msg.measure_rate_ms = MEASURE_RATE;
+    // require a minimum measurement rate of 5Hz
+    msg.measure_rate_ms = MIN(gps._rate_ms[state.instance], MINIMUM_MEASURE_RATE_MS);
```
Comment (OXINARF) [inline libraries/AP_GPS/AP_GPS_UBLOX.cpp]: Since this is in ms this should be a MAX right?

**Comment by WickedShell** [comment]
Nope, it's the number of milliseconds between readings so the larger the
number the fewer readings.

**Comment by WickedShell** [comment]
Alright, tested against a u-blox M8N, M8P, Neo-7N, LEA-6H. Interestingly they all behave a bit differently when using a update rate higher then requested. They all behave sanely however and will be detected as not working as long as the GPS config arming check is enabled, the default value should mean this is no change for normal use.

Also caught and corrected a case where we accepted a config as being correct to quickly, and a index that was off when reporting to the user.

**Comment by OXINARF** [comment]
LGTM

**Comment by magicrub** [comment]
@WickedShell  Is this still in DO NOT MERGE mode?

**Comment by WickedShell** [comment]
It's in sign of from @tridge stage. I'm otherwise happy with it but if he
could test it with the u-blox 5 on his desk that would be really valuable
to know if that works or not.

**Comment by tridge** [comment]
tested with a u-blox LEA 5H. It finds the GPS fine. I don't see any satellites due to my metal roof. Do you want me to test outside?

**Comment by WickedShell** [comment]
If you set the RATE_MS to 50 (which is an unreasonable value) does it pass
the GPS config part of the arming checks? (If not there is a one off debug
line I'd be interested in the result from)

**Comment by tridge** [comment]
applied, thanks Michael!

**Pull request closed without merging**